### PR TITLE
Update select columns

### PR DESCRIPTION
**Pull request by CWen001**
This PR resolves #(https://github.com/ericmjl/pyjanitor/issues/77).

# PR Checklist

Please ensure that you have done the following:

1. [x] PR in from a fork off your branch. Do not PR from <your_username>:master, but rather from <your_username>:<branch_name>.
2. [x] If you're not on the contributors list, add yourself to `AUTHORS.rst`.

## Quick Check

To do a very quick check that everything is correct, follow these steps below:

- [ ] Run the command `make check` from pyjanitor's top-level directory. This will automatically run:
    - black formatting
    - pycodestyle checkin
    - running the test suite
    - docs build

## Code Changes

If you are adding code changes, please ensure the following:

- [x] Ensure that you have added tests.
- [x] Run all tests (`$ pytest .`) locally on your machine.
    - [x] Check to ensure that test coverage covers the lines of code that you have added.
    - [x] Ensure that all tests pass.

## Documentation Changes

If you are adding documentation changes, please ensure the following:

- [ ] Build the docs locally.
- [ ] View the docs to check that it renders correctly.

# PR Description

Please describe the changes proposed in the pull request:

-  Add the functionality of selecting columns based on a shell-style glob string (e.g., `*_thing_*`)
-  Enable users to manually select several column names plus a pattern. (['a', 'b', 'col_*'])
-  For discussions please see https://github.com/ericmjl/pyjanitor/issues/77

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl

### CODE REVIEW

**Comment by CWen001** [comment]
Dear Eric, thanks for your kind instructions. I finally made this PR after trying to finish the PR Checklist. 

A few things to note. I found it difficult to use `make check` on my windows 10 64bit machine, and I googled some StackOverflow posts that said this command is absent on default windows environment. I'm not sure about this step. Please feel free to guide. The pytest results are `1 failed, 127 passed, 2 warnings in 32.75 seconds`. I do pass the tests for my edited select_columns, but it reported an error on `tests\functions\test_convert_unix_date.py `. The error message is `E   OverflowError: Python int too large to convert to C long`.

**Comment by ericmjl** [comment]
@CWen001 thanks for the PR! No worries about the `make check` command; this is a convenience command for Unix-like systems that have Make installed. 

You did the right thing by running all of the tests. The error message you encountered is probably just related to numerical precision on your machines, so I wouldn't worry about it.

If you click on the Travis CI details, you'll see that the error that shows up is related to code linting. Could you run `pycodestyle .` at your terminal?

I will also do a code review to see if there are other changes that need to happen. Other than that, super happy to see your PR in here!

File: janitor/functions.py [hunk 1]
```diff
@@ -2304,31 +2305,36 @@ def currency_column_to_numeric(
...
+            df = pd.DataFrame(...).select_columns(['a', 'b', 'col_*'], invert=True)
 
     :param df: A pandas DataFrame.
-    :param columns: A list of columns to select.
+    :param search_cols: A list of column names or search strings to be used to select. Valid inputs include
+        1) an exact column name to look for
+        2) a shell-style glob string (e.g., `*_thing_*`)
     :param invert: Whether or not to invert the selection.
-        This will result in selection ofthe complement of the columns provided.
+        This will result in selection of the complement of the columns provided.
```
Comment (ericmjl) [inline janitor/functions.py]: Great catch!

**Comment by CWen001** [comment]
Thanks! I ran the pycodestyle, and it shows following messages. Now may I ask which button I should click, please? I saw Changes requested but not sure what to do.
```
D:\GitHub\pyjanitor>activate pyjanitor-dev

(pyjanitor-dev) D:\GitHub\pyjanitor>pycodestyle .
.\janitor\functions.py:2319:80: E501 line too long (83 > 79 characters)
.\janitor\functions.py:2322:80: E501 line too long (107 > 79 characters)
.\janitor\functions.py:2326:80: E501 line too long (80 > 79 characters)
.\janitor\functions.py:2329:1: W293 blank line contains whitespace
.\janitor\functions.py:2337:80: E501 line too long (80 > 79 characters)
```

**Comment by ericmjl** [comment]
That's great! Parsing the messages by pycodestyle, you have 4 lines that are too long, and 1 line that contains blank whitespace.

There's two steps you can take here.

1. You can run black: `black -l 79 .`.
2. You can also fix the errors manually.  An example:

```
# Line 2319 has 83 characters, and is too long. 
.\janitor\functions.py:2319:80: E501 line too long (83 > 79 characters)
```

I'd suggest running black first, as it'll autofix most issues. The only tricky ones are the docstrings (if any), which you might end up manually editing.

**Comment by CWen001** [comment]
Aha, finally all checks passed! Thank you so much for letting me take this chance to learn. Actually, I just put things together without contributing too much, so really appreciate your encouragement and guide : )

**Comment by ericmjl** [comment]
Everything looks great to me! Thanks for the contribution, @CWen001 :smile:

**Pull request merged**